src/save.ts

```diff
@@ -27,7 +27,9 @@ async function run(): Promise<void> {
         const state = utils.getCacheState();
 
         // Inputs are re-evaluted before the post action, so we want the original key used for restore
-        const primaryKey = core.getState(State.CachePrimaryKey);
+        const primaryKey =
+            core.getState(State.CachePrimaryKey) || core.getInput(Inputs.Key);
```

Comment:
If I am not wrong, state value will be used by cache action and input will be used by save action. If so, then this statement is can be confusing as it is mixing logic for two actions.
We should have a pattern such that such action specific logic is separated out in an action specific wrapper.

Reply:
cc - @tanuj077